src/setup.ts

```diff
@@ -0,0 +1,47 @@
+import * as core from '@actions/core';
+import * as tc from '@actions/tool-cache';
+import {validateCheckSum, getNotationCheckSum} from './lib/checksum';
+import { getNotationDownloadURL } from './lib/install';
+
+// setup sets up the Notation CLI.
+async function setup(): Promise<void> {
+    try {
+        // inputs from user
+        const version: string = core.getInput('version');
+        const notation_url: string = core.getInput('url');
+        const notation_checksum = core.getInput('checksum').toLowerCase();
+
+        // sanity check
+        if (!(notation_url || version)) {
+            throw new Error("user needs to provide either version of official Notation CLI or url of customized Notation CLI release")
+        }
+        if (notation_url && !notation_checksum) {
+            throw new Error("user provided url of customized Notation CLI release but without SHA256 checksum")
+        }
+
+        // download Notation CLI and validate checksum
+        const downloadURL = getNotationDownloadURL(version, notation_url);
```

Comment:
Since you have included the download url in `notation_releases.json`, will getting download url form that file in the subsequent PR?

Reply:
Sure, we can add this logic in a following PR. Created an issue to track this: https://github.com/notaryproject/notation-action/issues/16